src/main/java/com/eternalcode/parcellockers/user/UserRepositoryJdbcImpl.java

```diff
@@ -101,19 +115,20 @@ public CompletableFuture<Void> remove(User user) {
 
     @NotNull
     private Optional<User> getUser(ResultSet resultSet) throws SQLException {
+        // TODO - Optimize this and use whenComplete() to get the parcels
         Set<Parcel> parcelSet = this.parcelRepository.findAll().join();
```

Comment:
w sumie, to user nie powinien mieć w sobie parcelów, tylko sobie komplikujesz wtedy wyciąganie danych